src/matoya.h

```diff
@@ -3339,13 +3339,16 @@ MTY_GetProcessDir(void);
 
 /// @brief Restart the current process.
 /// @details For more information, see `execv` from the C standard library.
+/// @param path Full path to the current process executable, or another executable to execv.
+///   May be set to `NULL` to restart the current process.
 /// @param argv Arguments to set up a call to `execv`. This is an array of strings
 ///   that must have its last element set to NULL.
+/// @param dir The working directory to set. May be NULL to use the current working directory
 /// @returns On success this function does not return, otherwise it returns false.
 ///   Call MTY_GetLog for details.
 //- #support Windows macOS Linux
 MTY_EXPORT bool
-MTY_RestartProcess(char * const *argv);
+MTY_RestartProcess(const char *path, const char * const *argv, const char *dir);
```

Comment:
I don't see any use in adding the path argument here, it will not really be "restart" at that point.

Reply:
The path argument is necessary for the PR this contributes to. Without it, the last step isn't possible